projects/plugins/social/src/class-rest-settings-controller.php

```diff
@@ -135,13 +135,19 @@ public function update_item( $request ) {
 					}
 					break;
 				case 'show_pricing_page':
-					if ( ! update_option( Jetpack_Social::JETPACK_SOCIAL_SHOW_PRICING_PAGE_OPTION, (int) $params[ $name ] ) ) {
```

Comment:
`update_option` returns `false` if the option wasn't updated (e.g. if you're setting it to `true` but it's already set to `true`), so we need to check the old value first before updating.

(or, alternatively, just get rid of the `WP_Error` here?)

Reply:
This is madness! Why doesn't it return WP_Error or false for failure and an int for option values updated.

Maybe we should just assume it did update?

Reply:
Yeah that's what I was considering as well. Done!